vsintegration/src/vs/FsPkgs/FSharp.Project/FS/AppConfigHelper.fs

```diff
@@ -304,115 +304,149 @@ namespace Microsoft.VisualStudio.FSharp.ProjectSystem
+                        //
+                        // The assembly version number scheme, evolved in a somewhat haphazard way:
+                        //  .Net 2 fsharp.core.dll has the version# 2.0.0.0 for VS2010 and 2.3.0.0 for VS2012 and up
+                        //  .Net 4 fsharp.core.dll has the version# 4.3.0.0 for VS2012, 4.3.1 for VS 2013 and 4.4.0.0
+                        //
+                        // Portable libraries are much different
+                        // There is a systematic scheme for portable libries from VS2013+Oob3.1.2 forward
+                        // It is:  3.Profile.Major.Minor
+                        //     Profile is .net framework profile number I.e 7,47,78 and 259
+                        //     Major.minor is the matching FSharp language major.minor version
+                        //
+                        // However in VS 2012 we released a Portable library which was profile 47 it has the version 2.3.5.0 in VS2013 it was updated to 2.3.5.1
+                        // and in VS 2013 we released an additional portable library based on the Windows 8 profile 7 with the version number 3.3.1.0
+                        //
+                        // Binding redirects are computed based on target fsharp core an fsharp core will be redirected to the target fsharp.core if it is "compatible"
+                        // Each desktop fsharp.core is a superset of the previous desktop fsharp.core dll's and is thus "compatible"
+                        // Each desktop fsharp.core.dll is also a superset of the portable libraries that shipped with it.
+                        // 
+                        // The table below represents the appropriate redirections
+                        // If the target version is between TagetMin and TargetMax inclusive then the redirects list contains the appropriate redirects
+                        //
+                        //TargetMin, targetMax, redirects
+                        "2.3.0.0",  "2.3.0.0",   ["2.0.0.0";   "2.0.0.0"]
+                        "2.3.5.1",  "2.3.5.1",   ["2.3.5.0";   "2.3.5.1"]
+                        "3.7.4.0",  "3.7.4.0",   ["3.3.1.0";   "3.7.4.0"]
+                        "3.47.4.0", "3.47.4.0",  ["2.3.5.0";   "2.3.5.1";   "3.47.4.0"]
+                        "3.78.4.0", "3.78.4.0",  ["3.78.3.1";  "3.78.4.0"]
+                        "3.259.4.0","3.259.4.0", ["3.259.3.1"; "3.259.4.0"]
+                        "4.3.0.0",  "4.4.0.0",   ["2.0.0.0";   "2.3.5.0";   "4.0.0.0";  "4.3.1.0"]
+                        "4.3.1.0",  "4.4.0.0",   ["3.3.1.0";   "2.3.5.1";   "3.78.3.1"; "3.259.3.1"]
+                        "4.4.0.0",  "4.4.0.0",   ["3.47.4.0";  "3.78.4.0";  "3.259.4.0"; "4.3.1.0";   "4.4.0.0"]
+                    ] |> Seq.where(fun (min, max, _) -> targetFSharpCoreVersion >= min && targetFSharpCoreVersion <= max)
```

Comment:
This is so crazy, but I think I get it.  A few places where I believe there are typos:

- First line should not have 2.0.0.0 twice (assume second one is supposed to be 2.3.0.0
- 4.3.0.0 line, need to add entry for 2.3.0.0, and last entry should be 4.3.0.0, not 4.3.1.0
- 4.3.1.0 line, need entry for 4.3.1.0
- 4.4.0.0 line, don't need 4.3.1.0 entry